### DISCUSSION TITLE
Is Datashare suitable for full-text indexing of SVN repositories?

### DISCUSSION THREAD

**Discussion by LeoGe2022** [ICIJ/datashare#1302] [Q&A]
I installed the Datashare in my desktop and found that most of my projects code (such as .cs, .vb, etc.) are not indexed. How can I configure the file extension types to be indexed? Thanks!

**Comment by pirhoo** [comment]
Hi @LeoGe2022, this is very surprising. Text files are obviously the most easy type to parse for Apache Tika. Do you see anything in the logs that might indicate what's happening?

I just tried with a few files and it worked without problem (Datashare 13.7.0):
![Screen Shot 2024-01-19 at 09 39 44](https://github.com/ICIJ/datashare/assets/471176/c7cd8763-badb-4de7-93d0-2b7c45f07ce9)

**Comment by LeoGe2022** [comment]
Hi @pirhoo,

Too much exceptions, I couldn't catch them all because the console scrolls up too fast.
I think that the Parse and Extract errors are normal, however, the following exceptions might be fatal:

```
org.elasticsearch.client.ResponseException: method [HEAD], host [http://localhost:9200], URI [/local-datashare/_doc/82fad08fa8cd6b9752c4602ceac16bfb74c3bda8a1718f597d8ba1dfaa8208c6bb4f9fbcf63572b9fbfb805b07dc0cec?stored_fields=_none_&_source=false], status line [HTTP/1.1 429 Too Many Requests]
```
```
java.lang.OutOfMemoryError: Java heap space
Exception in thread "elasticsearch[datashare][refresh][T#9]" java.lang.OutOfMemoryError: Java heap space
2024-01-19 20:14:39,923 [elasticsearch[datashare][refresh][T#10]] ERROR Engine - already closed by tragic event on the index writer

```
I suppose that some complex files such as ZIP, JAR cost too many memory, Actually I only need Datashare to index plain text and some MS Office files but I don't know how to select the file types.

Thanks!

**Reply by pirhoo** [reply]
It looks like your JVM is running out of memory which triggered a succession of failure.

How did you install Datashare? You might need to tweak the `-Xms` and `-Xmx` options.
